libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -63,6 +50,29 @@ class MapboxDirectionsSession(
         })
     }
 
+    override fun requestFasterRoute(
+        adjustedRouteOptions: RouteOptions,
+        routesRequestCallback: RoutesRequestCallback
+    ) {
+        if (routes.isEmpty()) {
+            return
+        }
```

Comment:
How about we call this method `requestAlternativeRoutes`? This would better describe what the method is actually doing and might be reused for other purposes in the future.

If we'd do that, we can move `if (routes.isEmpty())` check before invoking this method instead of having it here.

Just an idea, I'm fine with the current setup as well.

Reply:
The idea to reuse it for other purposes in the future sounds good to me. 

> Also, I would rather not use the term AlternativeRouteObserver as I feel that may be confusing from client's perspective. I say this because there is already this feature where we allow users to specify if they want alternative routes. and draw them accordingly. We then have offRoute observers and callbacks that inform users about onRoutesChanged. Having specific terms for features makes it easy to understand what is one doing. Wdyt?

I agree with @Guardiola31337 here. I wouldn't want to use the term `Alternative` here.
I would leave it as it is for now. Whenever we feel this can be reused we can revisit and think about renaming.